pkg/pusher/pusher.go

```diff
@@ -190,6 +198,21 @@ LOOP:
 						err = fmt.Errorf("pusher: receipt storer address: %w", err)
 						return
 					}
+
+					po := swarm.Proximity(ch.Address().Bytes(), storerPeer.Bytes())
+					d := s.depther.NeighborhoodDepth()
+					if po < d {
+						mtx.Lock()
```

Comment:
under this branch, you should also add the peer that returned the shallow-depth receipt to the skiplist of this particular chunk (not the protocol wide skiplist). This way, you will reduce the change of hitting exactly the same route again.

Reply:
Alternatively, the function that choses which peer to forward to, might be good to have a bit of randomness there and not deterministically choose always the same for each request.

Reply:
there are no skip lists nor even notions of peers in this component, so this is a bit out of scope. trying to get a different trace should be done tweaking the retry count and the pushsync protocol behavior by @esadakar